lib/cipherstash/client/rpc.rb

```diff
@@ -270,17 +270,23 @@ def put_stream(collection, records)
         # isn't the most idiomatic code on earth...
         requests = [[begin_request], records].lazy.flat_map { |x| x }
 
-        res = @metrics.measure_rpc_call("putStream") do
-          stub.put_stream(requests, metadata: rpc_headers)
-        end
+        num_inserted = 0
+        @logger.info("Cipherstash::Client::RPC#put_stream") { "Start streaming upsert of documents..." }
+
+        @metrics.measure_rpc_call("putStream") do
+          stub.put_stream_bi_direction(requests, metadata: rpc_headers).each do |res|
+            unless res.is_a?(Documents::StreamingPutReply)
+              raise Error::StreamingPutFailure, "expected Documents::StreamingPutReply response, got #{res.class} instead"
+            end
+            @logger.info("Cipherstash::Client::RPC#put_stream") { "#{res.numInserted} records inserted." }
 
-        unless res.is_a?(Documents::StreamingPutReply)
-          raise Error::StreamingPutFailure, "expected Documents::StreamingPutReply response, got #{res.class} instead"
+            num_inserted = res.numInserted
+          end
         end
 
-        raise_if_error(res)
```

Comment:
Do we still need to include this check? Can a `Documents::StreamingPutReply` contain an error?

Reply:
ah yes it could, I should have moved that up within the block. I'll add it back in.